### ISSUE TITLE
pattern uses more memory than 'maxmempattern'

### ISSUE THREAD

**Issue by anzawatta** [gu-fan/riv.vim#144]
Hi, and thanks for the maintenance.

I am getting the following error.
`Error detected while processing function riv#link#hi_hover[14]..riv#ptn#match_object:`
`E363: pattern uses more memory than 'maxmempattern'`

This error occurred by these lines in test.rst.
```
filepath1/filepath2/filepath3/filepath4/ab.png
filepath1filepath1/filepath3/sectio4/abcde.a
filepath1filepath1filepath1/filepath4/abcd.b
filepath1filepath1filepath1filepath1//abcdf.
```

**Comment by gu-fan** [comment]
well, I met this several times, maybe should change the regexp pattern

**Comment by goopyflux** [comment]
I'm getting the same error while using the `:figure:` directive with a filename that's a bit long.

**Comment by goopyflux** [comment]
Upon closer examination, it seems to have nothing to do with the `:figure:` directive, but with trying to highlight links up on hover(?). Would it be possible to just disable/toggle this feature through a setting option? Thanks.

**Comment by goopyflux** [comment]
Found this while reviewing the riv source:

> You can disable Cursor Highlighting by set '`g:riv_link_cursor_hl`_' to 0

At least there is a temporary work around. Thanks!

**Comment by gu-fan** [comment]
yes, it's caused by the link regexp pattern

**Comment by goopyflux** [comment]
I did a quick search for `E363: pattern uses more memory than 'maxmempattern'` and found [this](https://github.com/vim/vim/issues/2049). 

TL;DR Highlights:

- It's related to Vim and not specifically to the riv plugin.
- The default value for `maxmempattern` is 1000. So, one solution is to set it to something higher, like `set mmp=2000`. I have tried this and it seems to solve the issue. 
- The link above discusses other issues that might cause the `maxmempattern` error, including not having the latest Vim (duh!) and  Mac-specific issues. Might be worth a read after all.

Anyway, since it's not really a bug in Vim and has nothing to do with riv, perhaps it's best to close this issue(?).
